py/kubeflow/testing/util.py

```diff
@@ -93,6 +99,30 @@ def run(command,
 def run_and_output(*args, **argv):
   return run(*args, **argv)
 
+def combine_repos(list_of_repos):
+  """Builds a dictionary of repo owner/names to commit hashes.
+
+  Args:
+    list_of_repos: A list of repos to checkout, each one in the format of
+      "owner/name@commit".
+  Returns:
+    repos: A dictionary of repository names to commit hashes. Uses DEFAULT_REPOS as a base.
+  """
+
+  # 1. Convert list_of_repos to a dictionary
+  # where key is "repo_owner/repo_name" and value is the
+  # commit hash.
+  user_repos = {}
+  for r in list_of_repos:
+    parts = r.split('@')
+    user_repos[parts[0]] = parts[1]
+
+  # 2. Starting with the default repos, merge in any user-defined
+  # repos with commit hashes.
```

Comment:
Per comment above. I think the caller should pass in defaults because they will be different for each test.
Checking out repos can be expensive so we don't necessarily want to check out repos we don't need.